app/components/TimerButton.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from 'react'
+import { View, Text, StyleSheet } from 'react-native'
+import config, { colors, vectorIcons } from '../config'
+import { AnimatedCircularProgress } from 'react-native-circular-progress'
+import { useDispatch } from 'react-redux'
+import { addWashTime } from '../state/WashTimeHistory'
+
+const TimerButton = ({ navigation }) => {
+  const dispatch = useDispatch()
+  const timerDefault = config.timerDefault
+  const [timer, setCounter] = useState(timerDefault)
+  const [text, setText] = useState('Seconds')
+
+  const timerCounter = () => setCounter(timer - 1)
+
+  useEffect(() => {
+    if (timer <= 0) {
+      setText('Well Done')
+      dispatch(addWashTime({ datetime: Date.now() }))
+      return
+    }
+    const id = setInterval(timerCounter, 1000)
+    return () => {
+      clearInterval(id)
+    }
+  }, [timer])
+
+  const fill = 100 - timer * 5
```

Comment:
Magic numbers here. What would happen to the animation if we changed the defaultTimer time to 40 seconds?

Reply:
Okay then. What should i do?

Reply:
Can you have a think about a solution first?

Reply:
what you think about this : 

const fill = 100 - (timer * 100) / timerDefault

Reply:
yeah that looks good.